docs/src/main/asciidoc/security-architecture.adoc

```diff
@@ -57,7 +57,10 @@ For more information, see the xref:security-customization.adoc#security-identity
 
 == Supported authentication mechanisms
 
-To learn more about security authentication in Quarkus and the supported mechanisms and protocols, see the Quarkus xref:security-authentication-mechanisms.adoc[Authentication mechanisms in Quarkus] guide.
+The Quarkus Security framework supports multiple authentication mechanisms, which can also be combined.
+Some supported authentication mechanisms are built into Quarkus, while others require you to add an extension. 
+
+To learn about security authentication in Quarkus and the supported mechanisms and protocols, see the Quarkus xref:security-authentication-mechanisms.adoc[Authentication mechanisms in Quarkus] guide.
```

Comment:
Context: Added more detail and removed word 'more' because we haven't really learned anything yet reading this very high level overview.